management/connection_test.go

```diff
@@ -1,533 +1,143 @@
-				_, ok = c.Options.(*ConnectionOptionsApple)
-			case ConnectionStrategyLinkedin:
-				_, ok = c.Options.(*ConnectionOptionsLinkedin)
-			case ConnectionStrategyGitHub:
-				_, ok = c.Options.(*ConnectionOptionsGitHub)
-			case ConnectionStrategyWindowsLive:
-				_, ok = c.Options.(*ConnectionOptionsWindowsLive)
-			case ConnectionStrategySalesforce, ConnectionStrategySalesforceCommunity, ConnectionStrategySalesforceSandbox:
-				_, ok = c.Options.(*ConnectionOptionsSalesforce)
-			case ConnectionStrategyEmail:
-				_, ok = c.Options.(*ConnectionOptionsEmail)
-			case ConnectionStrategySMS:
-				_, ok = c.Options.(*ConnectionOptionsSMS)
-			case ConnectionStrategyOIDC:
-				_, ok = c.Options.(*ConnectionOptionsOIDC)
-			case ConnectionStrategyOAuth2:
-				_, ok = c.Options.(*ConnectionOptionsOAuth2)
-			case ConnectionStrategyAD:
-				_, ok = c.Options.(*ConnectionOptionsAD)
-			case ConnectionStrategyAzureAD:
-				_, ok = c.Options.(*ConnectionOptionsAzureAD)
-			case ConnectionStrategySAML:
-				_, ok = c.Options.(*ConnectionOptionsSAML)
-			case ConnectionStrategyGoogleApps:
-				_, ok = c.Options.(*ConnectionOptionsGoogleApps)
-			default:
-				_, ok = c.Options.(map[string]interface{})
-			}
-
-			if !ok {
-				t.Errorf("unexpected options type %T", c.Options)
-			}
-
-			t.Logf("%s %s %T\n", c.GetID(), c.GetName(), c.Options)
-		}
-	})
-
-	t.Run("Update", func(t *testing.T) {
-		id := c.GetID()
-
-		c.ID = nil       // read-only
-		c.Name = nil     // read-only
-		c.Strategy = nil // read-only
-
-		c.Options = &ConnectionOptions{
-
-			BruteForceProtection: auth0.Bool(true),
-			ImportMode:           auth0.Bool(false), // try some zero values
-			DisableSignup:        auth0.Bool(true),
-			RequiresUsername:     auth0.Bool(false),
-
-			CustomScripts:      map[string]interface{}{"get_user": "function( { return callback(null) }"},
-			Configuration:      map[string]interface{}{"foo": "bar"},
-			NonPersistentAttrs: &[]string{"ethnicity", "gender"},
-		}
-
-		err = m.Connection.Update(id, c)
-		if err != nil {
-			t.Error(err)
-		}
-
-		t.Logf("%v\n", c)
-	})
-
-	t.Run("Delete", func(t *testing.T) {
-		err = m.Connection.Delete(c.GetID())
-		if err != nil {
-			t.Error(err)
-		}
-	})
-
-	t.Run("ReadByName", func(t *testing.T) {
-		cs, err := m.Connection.ReadByName("Username-Password-Authentication")
-		if err != nil {
-			t.Error(err)
-		}
-		t.Logf("%v\n", cs)
-	})
-
-	t.Run("ReadByNameEmptyName", func(t *testing.T) {
-		cs, err := m.Connection.ReadByName("")
-		if err == nil {
-			t.Fail()
-		}
-		mgmtError, ok := err.(*managementError)
-		if !ok {
-			t.Fail()
-		}
-		if mgmtError.StatusCode != 400 {
-			t.Fail()
-		}
-		if cs != nil {
-			t.Fail()
-		}
-		t.Logf("%v\n", cs)
-	})
-}
-
-func TestConnectionOptions(t *testing.T) {
-	t.Run("GoogleOAuth2", func(t *testing.T) {
-		g := &Connection{
-			Name:     auth0.Stringf("Test-Connection-%d", time.Now().Unix()),
+var connectionTestCases = []connectionTestCase{
```

Comment:
Adding all connection types at the top so we can reuse them in all the CRUD func tests and expand coverage as well.